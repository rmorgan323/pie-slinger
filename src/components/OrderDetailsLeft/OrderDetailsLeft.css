/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 20px;
  font-family: "Century Gothic", CenturyGothic, sans-serif;
  outline: none;
  color: #8b5e3c;
  box-sizing: border-box; }

h1 {
  font-family: "tekton-pro", sans-serif;
  line-height: 1.3; }

h2, h3, h4, h5, h6 {
  font-family: "Century Gothic", CenturyGothic, sans-serif;
  line-height: 1.3;
  font-weight: 500; }

h3 {
  color: #8b5e3c;
  font-weight: 900;
  margin-top: 10px;
  line-height: 1.5; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 39px; }

h3 {
  font-size: 34px; }

h4 {
  font-size: 27px; }

h5 {
  font-size: 20px; }

h6 {
  font-size: 12px; }

p {
  font-size: 20px;
  line-height: 1.3;
  font-weight: 300;
  color: #d1be95; }

/***  MIXINS  ***/
.OrderDetailsLeft {
  padding: 0 30px;
  width: 260px; }
  .OrderDetailsLeft .order-container {
    border-left: 6px solid #ded2bb;
    padding-left: 10px; }
  .OrderDetailsLeft span {
    color: #d1be95; }
  .OrderDetailsLeft .font-bold {
    color: #8b5e3c; }

@media screen and (max-width: 919px) {
  .OrderDetailsLeft {
    margin: auto; } }
