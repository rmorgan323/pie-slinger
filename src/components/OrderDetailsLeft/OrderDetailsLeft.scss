@import '../../variables.scss';

.OrderDetailsLeft {
  width: 260px;
  padding: 0 30px;

  .order-container {
    @include border-left;
  }
}