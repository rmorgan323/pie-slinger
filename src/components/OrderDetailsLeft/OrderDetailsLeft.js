import React from 'react';
import { PropTypes } from 'prop-types';
import moment from 'moment';
import cap from '../../helper/cap/cap';
import checkNormal from '../../helper/checkNormal/checkNormal';
import './OrderDetailsLeft.css';

const OrderDetailsLeft = ({ date, orderId, crust, size }) => {

  return (
    <div className="OrderDetailsLeft">
      <div>
        <h3>ORDER</h3>
        <div className="order-container">
          <p>{moment(date).format('MMMM Do, h:mm a')}</p>
          <p>Order ID: {orderId}</p>
        </div>
        <h3>PIZZA</h3>
        <div className="order-container">
          <p>Crust: <span className={checkNormal(crust)}>{cap(crust)}</span></p>
          <p>Size: <span className={checkNormal(size)}>{cap(size)}</span></p>
        </div>
      </div>
    </div>
  );
};

export default OrderDetailsLeft;

OrderDetailsLeft.propTypes = {
  date: PropTypes.string,
  orderId: PropTypes.number,
  crust: PropTypes.string,
  size: PropTypes.string
};