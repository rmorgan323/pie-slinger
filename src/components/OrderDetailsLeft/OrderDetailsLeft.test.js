import React from 'react';
import { shallow } from 'enzyme';
import OrderDetailsLeft from './OrderDetailsLeft';

const mockPizza = {
  date: "2018-04-25T06:32:31.259Z",
  orderId: 9273479,
  crust: "thin",
  size: "medium"
};

describe('OrderDetailsLeft tests', () => {
  it('should match the snapshot', () => {
    const renderedApp = shallow(
      <OrderDetailsLeft 
        date={mockPizza.date} 
        orderId={mockPizza.orderId} 
        crust={mockPizza.crust}
        size={mockPizza.size}  
      />
    );
    expect(renderedApp).toMatchSnapshot();
  });
});