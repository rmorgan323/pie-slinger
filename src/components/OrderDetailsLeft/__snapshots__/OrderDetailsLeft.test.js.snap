// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetailsLeft tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderDetailsLeft
    crust="thin"
    date="2018-04-25T06:32:31.259Z"
    orderId={9273479}
    size="medium"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h3>
                ORDER
        </h3>
        <div
                className="order-container"
        >
                <p>
                        April 25th, 12:32 am
                </p>
                <p>
                        Order ID: 
                        9273479
                </p>
        </div>
        <h3>
                PIZZA
        </h3>
        <div
                className="order-container"
        >
                <p>
                        Crust: 
                        <span
                                className="font-bold"
                        >
                                Thin
                        </span>
                </p>
                <p>
                        Size: 
                        <span
                                className={undefined}
                        >
                                Medium
                        </span>
                </p>
        </div>
</div>,
      "className": "OrderDetailsLeft",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            ORDER
</h3>,
          <div
            className="order-container"
>
            <p>
                        April 25th, 12:32 am
            </p>
            <p>
                        Order ID: 
                        9273479
            </p>
</div>,
          <h3>
            PIZZA
</h3>,
          <div
            className="order-container"
>
            <p>
                        Crust: 
                        <span
                                    className="font-bold"
                        >
                                    Thin
                        </span>
            </p>
            <p>
                        Size: 
                        <span
                                    className={undefined}
                        >
                                    Medium
                        </span>
            </p>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ORDER",
          },
          "ref": null,
          "rendered": "ORDER",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                April 25th, 12:32 am
</p>,
              <p>
                Order ID: 
                9273479
</p>,
            ],
            "className": "order-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "April 25th, 12:32 am",
              },
              "ref": null,
              "rendered": "April 25th, 12:32 am",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Order ID: ",
                  9273479,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Order ID: ",
                9273479,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PIZZA",
          },
          "ref": null,
          "rendered": "PIZZA",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Crust: 
                <span
                                className="font-bold"
                >
                                Thin
                </span>
</p>,
              <p>
                Size: 
                <span
                                className={undefined}
                >
                                Medium
                </span>
</p>,
            ],
            "className": "order-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Crust: ",
                  <span
                    className="font-bold"
>
                    Thin
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Crust: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Thin",
                    "className": "font-bold",
                  },
                  "ref": null,
                  "rendered": "Thin",
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Size: ",
                  <span
                    className={undefined}
>
                    Medium
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Size: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Medium",
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": "Medium",
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h3>
                    ORDER
          </h3>
          <div
                    className="order-container"
          >
                    <p>
                              April 25th, 12:32 am
                    </p>
                    <p>
                              Order ID: 
                              9273479
                    </p>
          </div>
          <h3>
                    PIZZA
          </h3>
          <div
                    className="order-container"
          >
                    <p>
                              Crust: 
                              <span
                                        className="font-bold"
                              >
                                        Thin
                              </span>
                    </p>
                    <p>
                              Size: 
                              <span
                                        className={undefined}
                              >
                                        Medium
                              </span>
                    </p>
          </div>
</div>,
        "className": "OrderDetailsLeft",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              ORDER
</h3>,
            <div
              className="order-container"
>
              <p>
                            April 25th, 12:32 am
              </p>
              <p>
                            Order ID: 
                            9273479
              </p>
</div>,
            <h3>
              PIZZA
</h3>,
            <div
              className="order-container"
>
              <p>
                            Crust: 
                            <span
                                          className="font-bold"
                            >
                                          Thin
                            </span>
              </p>
              <p>
                            Size: 
                            <span
                                          className={undefined}
                            >
                                          Medium
                            </span>
              </p>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ORDER",
            },
            "ref": null,
            "rendered": "ORDER",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  April 25th, 12:32 am
</p>,
                <p>
                  Order ID: 
                  9273479
</p>,
              ],
              "className": "order-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "April 25th, 12:32 am",
                },
                "ref": null,
                "rendered": "April 25th, 12:32 am",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Order ID: ",
                    9273479,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Order ID: ",
                  9273479,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PIZZA",
            },
            "ref": null,
            "rendered": "PIZZA",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Crust: 
                  <span
                                    className="font-bold"
                  >
                                    Thin
                  </span>
</p>,
                <p>
                  Size: 
                  <span
                                    className={undefined}
                  >
                                    Medium
                  </span>
</p>,
              ],
              "className": "order-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Crust: ",
                    <span
                      className="font-bold"
>
                      Thin
</span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Crust: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Thin",
                      "className": "font-bold",
                    },
                    "ref": null,
                    "rendered": "Thin",
                    "type": "span",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Size: ",
                    <span
                      className={undefined}
>
                      Medium
</span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Size: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Medium",
                      "className": undefined,
                    },
                    "ref": null,
                    "rendered": "Medium",
                    "type": "span",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
